docs/concepts/what-is-a-failure.md

```diff
@@ -0,0 +1,137 @@
+---
+id: what-is-a-failure
+title: What is a Failure?
+sidebar_label: Failure
+description: A Failure is Temporal's representation of various types of errors that occur in the system.
```

Comment:
Thinking about this -
I want to see this moved into the KB -
Title is OK as is - or consider something like "What do we mean by Failure?"

Reply:
Can we have a KB article included in the Glossary? (Failure is a temporal term that I'd expect to find there)

Reply:
No not at the moment.
But, I am failing to see how Failure is being defined as Temporal specific here.

> A Failure is Temporal's representation of various types of errors that occur in the system.

That doesn't feel super concrete.
I feel like what we really need to do is either:

- Define each status clearly for both an Activity Execution and a Workflow Execution within the "what is" topic respectfully.
- OR - have a concept page for each status for each Execution type.

In either situation we would want to explain what it means to be in that status and how it could come about.
Particularly, what errors one might see if in that status.

I think one problem with trying to define "Failure" as a core term, is that our executions can encounter a failure - but still be trying to progress.
Basically this page tries to define Failure as a the full range of final execution failures, failure but still possible to progress, and failure failure (u really effed up out of the context of development).

So I would expect to define Temporal Execution statuses, execution errors, error types - any of those as core terms potentially. At this point, I feel pretty strongly that this is not a core term.

BUT - I know of someone we can ask who can set us straight. @dtornow do you think Failure fits as a core term for Temporal. Or is it something different?

Reply:
OK, so chatting with @dtornow - you are actually totally on point in expecting a "Failure" concept explanation - I believe it would look at lot more like this: https://dominik-tornow.medium.com/handling-failures-from-first-principles-1ed976b1b869

So, this content looks a lot more like a Failure reference than anything at this point - which I would then propose is more aligned with the KB or references section of the docs.

Reply:
The main issue I think we need to make clear is how erroring out of a workflow is treated and how erroring out of an activity is treated. They are different and surprising and important. The failure concept, references of types of failures, KBs about best practices, etc should all take a back seat to this important documentation.

Reply:
Moved content to `/references`, and made a short helpful-for-now concept page:

- [/references/failures](https://temporal-documentation-git-failures.preview.thundergun.io/references/failures)
- [/concepts/what-is-a-failure](https://temporal-documentation-git-failures.preview.thundergun.io/concepts/what-is-a-failure)

> we need to make clear is how erroring out of a workflow is treated and how erroring out of an activity is treated. They are different and surprising and important.

That's included in the Application Failure section, and I imagine will be clear in the forthcoming section of the dev guide (cc @rachfop)

[/references/failures#application-failure](https://temporal-documentation-git-failures.preview.thundergun.io/references/failures#application-failure)